lib/logstash/inputs/heartbeat.rb

```diff
@@ -43,23 +47,27 @@ def run(queue)
     sequence = 0
 
     Stud.interval(@interval) do
-      if @message == "epoch"
-        event = LogStash::Event.new("clock" => Time.now.to_i, "host" => @host)
-      elsif @message == "sequence"
-        event = LogStash::Event.new("clock" => sequence, "host" => @host)
-        sequence += 1
-      else
-        event = LogStash::Event.new("message" => @message, "host" => @host)
-      end
-      
+      sequence += 1
+      event = generate_message(sequence)
       decorate(event)
       queue << event
-
+      break if sequence == @count
     end # loop
 
   end # def run
 
   public
+  def generate_message(sequence)
+    if @message == "epoch"
+      LogStash::Event.new("clock" => Time.now.to_i, "host" => @host)
+    elsif @message == "sequence"
+      LogStash::Event.new("clock" => sequence, "host" => @host)
+    else
+      LogStash::Event.new("message" => @message, "host" => @host)
+    end
+  end
```

Comment:
I agree that it's a yak shave.  I think that'd be more than needed for this lightweight plugin.